### DISCUSSION TITLE
UNDIONLY.KPXE cannot execute commands written in iPXE

### DISCUSSION THREAD

**Discussion by wkchong2000** [ipxe/ipxe#392] [Q&A]
When my system is booting up in UNDIONLY.KPXE and loading the "default.ipxe" menu that I created for IPXE.EFI, it failed to execute "menu" command.  
error message:
"menu: command not found
Could not boot: Exec format error (http://ipxe.org/2e022001)"

My question is: Can I use the same iPXE.EFI menu for UNDIONLY.KPXE menu?

Here is the contents of my default.ipxe file

#!ipxe
set base http://192.168.1.5/netboot

########################## MAIN MENU #############################
:MAIN
menu TS Lab iPXE Test Server
item
item OS   OS Installation
item shell      iPXE Shell
item return     Refresh
choose target && goto ${target}

**Comment by NiKiZe** [comment]
Yes it is the same, you can try to do an imgfetch on the script and do imgstat, that should show you that it is a script.

If not it's probably due to not starting with #!ipxe for example having BOM chars or similar.
